flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcGateway.java

```diff
@@ -34,4 +34,16 @@ public interface RpcGateway {
      * @return Fully qualified hostname under which the associated rpc endpoint is reachable
      */
     String getHostname();
+
+    /**
+     * Perform optional steps before handling any messages or performing service actions (e.g.
+     * start/stop).
+     */
+    default void beforeInvocation() {}
```

Comment:
Not a fan of these because it means we couple components more. I'd rather see us add some pre/post-runnable args to wherever the MDC is set up.

Reply:
Do you mean passing these Runnables to `actor.Props.create()`?

Reply:
yes, (if this is handled in the PekkoRpcActor). Although maybe not runnables but just a Map<String, String>? Not sure if we _really_ want to open us up to running arbitrary things before/after just yet 🤔

Reply:
Yeah, I'd rather keep it as an explicit method.